snuba/optimize.py

```diff
@@ -128,19 +253,29 @@ def get_partitions_to_optimize(
     return parts
 
 
-def _subdivide_parts(
-    parts: Sequence[util.Part], number_of_subdivisions: int
-) -> Sequence[Sequence[util.Part]]:
+def _subdivide_partitions(
+    partitions: Sequence[str], number_of_subdivisions: int
+) -> Sequence[Sequence[str]]:
     """
-    Subdivide a list of parts into number_of_subdivisions lists of parts so that optimize can be executed
-    on different parts list by different threads.
+    Subdivide a list of partitions into number_of_subdivisions lists of partitions
+    so that optimize can be executed on different partitions list by different
+    threads.
     """
 
-    sorted_parts = sorted(parts, key=lambda part: part.date, reverse=True)
-    output: MutableSequence[Sequence[util.Part]] = []
+    def sort_partitions(partition_name: str) -> str:
```

Comment:
Why sorting at all ?

Reply:
Sorting the partitions since I want to ensure that the most recent partitions are optimized before the older ones. Our product mostly serves queries which are more recent. If we ever hit the timeout condition this would ensure that the queries for recent data would not have issues.